nibabel/orientations.py

```diff
@@ -350,6 +350,7 @@ def axcodes2ornt(axcodes, labels=None):
            [ 2.,  1.]])
     """
     if labels is None:
+        axcodes = tuple(c.upper() if c is not None else None for c in axcodes)
```

Comment:
Should this be run only when `labels is None`?

Reply:
Yes. Unless the labels themselves are also to be filtered. See https://github.com/nipy/nibabel/pull/590#discussion_r163340494.

Reply:
Doesn't that smell off though - to be case insensitive only when the labels are not specified?

I guess the ideal would be to be case insensitive only for codes labels in `LRAPIS`.

Reply:
I see, so the issue is if someone passes an alternative default ordering of the normal codes, but not custom labels, we want the overall behavior to be unchanged. I was under the impression that the only time you wanted to specify labels would be to use some custom labeling.

Then yeah, that might make sense. What if it becomes:

```Python
if labels is None:
    labels = list(zip('LPI', 'RAS'))

# Enable case-insensitive axis codes for standard direction labels
if set(elem for pair in labels for elem in pair) == set('LRAPIS'):
    axcodes = tuple(...)
```

Reply:
@matthew-brett Can you please confirm whether I've summarized your concern and proposed a reasonable solution? If so, I think we can get this fix in pretty soon.

Reply:
Sorry to be a bit imprecise.

Now I think about this more, it still seems ugly to me to special case for characters `LRAPIS`.  Why these and not other codes?   For example, what if someone uses `F` and `B` for front and back?  Wouldn't they expect the same behavior?

I've very sorry to keep changing, but can't we go back to expecting upper case for the codes from the caller, and, in particular, fixing this in the trackvis caller rather than here?

Reply:
@matthew-brett would it make sense to put in a "header format check" that returns a more informative error? That way the user can see why the load is failing and modify the header themself?